File: src/main/java/com/perksy/kafkacodeexercise/ResultController.java
package com.perksy.kafkacodeexercise;

import com.perksy.kafkacodeexercise.model.OrderResult;
import org.apache.kafka.streams.KafkaStreams;
import org.apache.kafka.streams.StoreQueryParameters;
import org.apache.kafka.streams.state.QueryableStoreTypes;
import org.springframework.beans.factory.ObjectProvider;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ResultController {
    private ObjectProvider<KafkaStreams> streamsProvider;

    public ResultController(ObjectProvider<KafkaStreams> streamsProvider) {
        this.streamsProvider = streamsProvider;
    }

    @GetMapping("/orderresult/{id}")
    OrderResult one(@PathVariable String id) {
        return streamsProvider.getObject().store(
                StoreQueryParameters.fromNameAndType("ResultsStore", QueryableStoreTypes.<String, OrderResult>keyValueStore())
        ).get(id);
    }
}

File: src/main/kotlin/com/perksy/kafkacodeexercise/ResultController.kt
package com.perksy.kafkacodeexercise

import com.perksy.kafkacodeexercise.model.OrderResult
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class ResultController() {

    @GetMapping("/orderresult/{id}")
    fun one(@PathVariable id: String): OrderResult? {
        return null
    }
}